app/assets/javascripts/Components/admin_users_list.jsx

```diff
@@ -0,0 +1,97 @@
+import React from "react";
+import {render} from "react-dom";
+import ReactTable from "react-table";
+import {selectFilter} from "./Helpers/table_helpers";
+
+class AdminUsersList extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      users: [],
+      loading: true,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  fetchData = () => {
+    $.ajax({
+      url: Routes.admin_users_path(),
+      dataType: "json",
+    }).then(data => {
+      this.setState({users: data, loading: false});
+    });
+  };
+
+  columns = userTypes => [
+    {
+      Header: I18n.t("activerecord.attributes.user.user_name"),
+      accessor: "user_name",
+      id: "user_name",
+      minWidth: 120,
+    },
+    {
+      Header: I18n.t("activerecord.attributes.user.first_name"),
+      accessor: "first_name",
+      minWidth: 120,
+    },
+    {
+      Header: I18n.t("activerecord.attributes.user.last_name"),
+      accessor: "last_name",
+      minWidth: 120,
+    },
+    {
+      Header: I18n.t("activerecord.attributes.user.email"),
+      accessor: "email",
+      minWidth: 150,
+    },
+    {
+      Header: I18n.t("activerecord.attributes.user.id_number"),
+      accessor: "id_number",
+      minWidth: 90,
+      className: "number",
+    },
+    {
+      Header: I18n.t("activerecord.attributes.user.user_type"),
+      accessor: "type",
+      minWidth: 90,
+      filterMethod: (filter, row) => {
+        if (filter.value === "all") {
+          return true;
+        } else {
+          return filter.value === row[filter.id];
+        }
+      },
+      Filter: selectFilter,
+      filterOptions: userTypes.map(type => {
```

Comment:
Eh, I don't think you need to make this a parameter to the function; a class constant is more appropriate since these values won't change.

Also for the display values, you should use internationalized strings (`I18n`) for the model names for `AdminUser` and `EndUser` (and in particular, add a space between the words).